packages/graphql/src/schema/validation/custom-rules/warnings/list-of-lists.ts

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (c) "Neo4j"
+ * Neo4j Sweden AB [http://neo4j.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Kind } from "graphql";
+import type { ListTypeNode, NonNullTypeNode, ASTVisitor, FieldDefinitionNode } from "graphql";
+
+export function WarnIfListOfListsFieldDefinition(): ASTVisitor {
+    let warningAlreadyIssued = false;
+
+    return {
+        FieldDefinition(field: FieldDefinitionNode) {
+            if (!warningAlreadyIssued) {
+                const listTypeNode = getListTypeNode(field);
+
+                if (listTypeNode) {
+                    if (getListTypeNode(listTypeNode)) {
+                        console.warn(
+                            "Encountered list field definition(s) with list elements. This is not supported by Neo4j, however, you can ignore this warning if the field is only used in the result of custom resolver/Cypher."
+                        );
+                        warningAlreadyIssued = true;
+                    }
+                    // const isListOfLists = listNode.type.kind === Kind.LIST_TYPE;
+                    // // TODO: figure this out - seems to have no impact having this commented-out
+                    // // && listNode.type.type.kind === Kind.LIST_TYPE;
+                    // if (isListOfLists) {
+                    //     console.warn(
+                    //         "Encountered list field definition(s) with list elements. This is not supported by Neo4j, however, you can ignore this warning if the field is only used in the result of custom resolver/Cypher."
+                    //     );
+                    //     warningAlreadyIssued = true;
+                    // }
```

Comment:
is this comment useful?

Reply:
It was already there in the original implementation! 😆 But I've removed it now.